src/client.rs

```diff
@@ -209,16 +211,19 @@ pub unsafe extern "C" fn native_stats_cb<C: Context>(
         _conf: *mut RDKafka, json: *mut i8, json_len: usize,
         opaque: *mut c_void
 ) -> i32 {
-    let json_string = String::from_raw_parts(json as *mut u8, json_len, json_len);
-
     let context = Box::from_raw(opaque as *mut C);
-    (*context).stats(json_string);
+
+    let statistics_json = String::from_raw_parts(json as *mut u8, json_len, json_len);
+    match serde_json::from_str(&statistics_json) {
+        Ok(stats) => (*context).stats(stats),
+        Err(e) => error!("Could not parse statistics json: {}", e)
+    }
+
     mem::forget(context);   // Do not free the context
 
     1 // librdkafka will not free the pointer
```

Comment:
I'm not sure what the point of this return code is, I only changed it up to line 221.